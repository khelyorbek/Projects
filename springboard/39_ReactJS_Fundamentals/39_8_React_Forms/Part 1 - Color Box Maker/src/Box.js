import React from 'react';

// this component should display a div with a background color, width and height based on the props passed to it.
const Box = (props) => {
    const handleRemove = (e) => {
        // console.log(e.target.bgColor);
        // console.log(props.bgColor);
        props.removeBox(props.bgColor);
    }

    return (<div style={{display: 'flex', fontSize: '3rem', alignItems: 'center', justifyContent: 'center', textAlign: 'center', width: `${props.width}px`, height: `${props.height}px`, backgroundColor: props.bgColor}} onClick={handleRemove}>REMOVE</div>)
}

export default Box;