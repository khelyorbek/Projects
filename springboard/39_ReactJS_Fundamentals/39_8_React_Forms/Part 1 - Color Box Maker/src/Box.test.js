import React from 'react';
import { render } from "@testing-library/react";
import Box from './Box';

// smoke test
it('renders without crashing', function() {
  render(<Box />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

