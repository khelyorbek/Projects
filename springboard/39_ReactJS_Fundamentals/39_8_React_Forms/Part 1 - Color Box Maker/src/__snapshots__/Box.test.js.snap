// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; font-size: 3rem; align-items: center; justify-content: center; text-align: center;"
  >
    REMOVE
  </div>
</DocumentFragment>
`;
